src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -225,50 +212,19 @@ internal async Task ExecuteBatchSafeAsync(TableBatchOperation batch, DateTime? n
             }
         }
 
-        internal async Task CreateIfNotExistsAsync(CloudTable table, int retryDelayMS = 1000)
+        internal async Task CreateIfNotExistsAsync(TableClient table, int retryDelayMS = 1000)
         {
-            int attempt = 0;
-            do
-            {
-                try
-                {
-                    bool tableCreated = true;
-                    if (!table.Exists())
-                    {
-                        tableCreated = await table.CreateIfNotExistsAsync();
-                    }
+            bool tableCreated = await TableStorageHelpers.CreateIfNotExistsAsync(table, TableServiceClient, _tableCreationRetries, retryDelayMS);
 
-                    if (tableCreated && _scaleOptions.MetricsPurgeEnabled)
-                    {
-                        // when we roll over to a new table, it's a good time to
-                        // do a background purge of any old tables
-                        QueueBackgroundMetricsTablePurge();
-                    }
-                }
-                catch (StorageException e)
-                {
-                    // Can get conflicts with multiple instances attempting to create
-                    // the same table.
-                    // Also, if a table queued up for deletion, we can get a conflict on create,
-                    // though these should only happen in tests not production, because we only ever
-                    // delete OLD tables and we'll never be attempting to recreate a table we just
-                    // deleted outside of tests.
-                    if (e.RequestInformation.HttpStatusCode == (int)HttpStatusCode.Conflict &&
-                        attempt < _tableCreationRetries)
-                    {
-                        // wait a bit and try again
-                        await Task.Delay(retryDelayMS);
-                        continue;
-                    }
-                    throw;
-                }
-
-                return;
+            if (tableCreated && _scaleOptions.MetricsPurgeEnabled)
+            {
+                // when we roll over to a new table, it's a good time to
+                // do a background purge of any old tables
+                QueueBackgroundMetricsTablePurge();
```

Comment:
Based on the implementation of `QueueBackgroundMetricsTablePurge`, I don't really see how the previous exception handling logic was ever engaged.

Reply:
This logic hasn't change with the migration, It'll log whatever error and stop processing. On the worst case-scenario, it might also happen that if the issue is not solved, it never purges old tables, filling the storage account. Shall we flag this for a future refactor or did you have something else in mind for this PR?

Reply:
No action required here.